src/main/java/org/reaktivity/nukleus/kafka/internal/stream/NetworkConnectionPool.java

```diff
@@ -368,44 +343,18 @@ private void doAttach(
             onMetadataError.accept(errorCode);
             break;
         case NONE:
-            // For streaming topics default to receiving only live (new) messages
-            final long defaultOffset = fetchOffsets.isEmpty() && !topicMetadata.compacted ? MAX_OFFSET : 0L;
-
-            if (fetchKey != null)
-            {
-                int partitionId = BufferUtil.partition(partitionHash, topicMetadata.partitionCount());
-                long offset = fetchOffsets.computeIfAbsent(0L, v -> defaultOffset);
-                long lowestOffset = topicMetadata.firstAvailableOffset(partitionId);
-                offset = Math.max(offset,  lowestOffset);
-                if (partitionId != 0)
-                {
-                    fetchOffsets.remove(0L);
-                }
-                fetchOffsets.put(partitionId, offset);
-            }
-            else
-            {
-                final int partitionCount = topicMetadata.partitionCount();
-                for (int partition=0; partition < partitionCount; partition++)
-                {
-                    long offset = fetchOffsets.computeIfAbsent(partition, v -> defaultOffset);
-                    long lowestOffset = topicMetadata.firstAvailableOffset(partition);
-                    offset = Math.max(offset,  lowestOffset);
-                    fetchOffsets.put(partition, offset);
-                }
-            }
             final NetworkTopic topic = topicsByName.computeIfAbsent(topicName,
```

Comment:
Verify no loss of semantics caused by above removal.

Reply:
This code was moved into ClientAcceptStream.onAttachPrepared. So no loss of semantics.